src/app/search/state/search.service.ts

```diff
@@ -75,6 +76,38 @@ export class SearchService {
     private imageProviderService: ImageProviderService
   ) {}
 
+  /**
+   * Return -1 if a sorts before b, 1 if b sorts before a, and 0 if they are the same,
+   * comparing based on the given attribute and direction
+   * @param a: DockstoreTool or Workflow
+   * @param b: DockstoreTool or Workflow
+   * @param attribute: workflow or tool property to sort by
+   * @param direction: 'asc' or 'desc'
+   */
+  static compareAttributes(a: DockstoreTool | Workflow, b: DockstoreTool | Workflow, attribute: string, direction: string) {
+    let aVal = a[attribute];
+    let bVal = b[attribute];
+
+    // regardless of sort direction, null or empty values go at the end
+    if (!aVal) {
+      return 1;
+    }
+    if (!bVal) {
+      return -1;
+    }
+
+    // ignore case when sorting by author or name
+    if (attribute === 'author' || attribute === 'name') {
+      aVal = aVal.toLowerCase();
+      bVal = bVal.toLowerCase();
+    }
+    if (direction === 'asc') {
+      return aVal < bVal ? -1 : aVal > bVal ? 1 : 0;
```

Comment:
I missed this the first time, but I think you should be using [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare). Something like `aVal.localCompare(bVal)`, and then multiple by -1 if desc

Reply:
ah okay, that does look cleaner